patterns/callback/callbacker/src/types/pending_callback_note.nr

```diff
@@ -0,0 +1,183 @@
+/// @member beneficiary: The address of the callback beneficiary (Alice)
+/// @member data: The data associated to use for the callback
+/// @member header: The note header.
+struct PendingCallbackNote {
+    owner: AztecAddress,
+    sender: AztecAddress,
+    beneficiary: AztecAddress,
+    data: Field,
+    header: NoteHeader
+}
+
+/// @notice The length of the pending callback note excluding the header
+global PENDING_CALLBACK_NOTE_LEN: Field = 4;
+
+/// @notice Implements the methods of the PendingCallbackNote
+impl PendingCallbackNote {
+    /// @notice Creates a new PendingCallbackNote
+    /// @param owner The address of the note owner
+    /// @param sender The address of the callback request sender
+    /// @param beneficiary The address of the beneficiary
+    /// @param data The data associated with the note
+    /// @return A new instance of PendingCallbackNote
+    pub fn new(
+            owner: AztecAddress,
+            sender: AztecAddress,
+            beneficiary: AztecAddress,
+            data: Field,
+        ) -> Self {
+
+        PendingCallbackNote {
+            owner,
+            sender,
+            beneficiary,
+            data,
+            header: NoteHeader::empty(),
+        }
+
+    }
+
+    /// @notice Serializes the note into a preimage.
+    /// @return An array of fields representing the serialized note.
+    pub fn serialize(self) -> [Field; PENDING_CALLBACK_NOTE_LEN] {
+        [
+            self.owner.to_field(),
+            self.sender.to_field(),
+            self.beneficiary.to_field(),
+            self.data,
+        ]
+    }
+
+    /// @notice Deserializes a note from a preimage.
+    /// @param preimage An array of fields representing the serialized note.
+    /// @return A new instance of PendingCallbackNote based on the provided preimage.
+    pub fn deserialize(preimage: [Field; PENDING_CALLBACK_NOTE_LEN]) -> Self {
+        PendingCallbackNote {
+            owner: AztecAddress::from_field(preimage[0]),
+            sender: AztecAddress::from_field(preimage[1]),
+            beneficiary: AztecAddress::from_field(preimage[2]),
+            data: preimage[3],
+            header: NoteHeader::empty(),
+        }
+    }
+
+    /// @notice Computes the hash of the note.
+    /// @dev Utilizes the Pedersen hash function to compute the hash of the serialized note.
+    /// @return The computed hash of the note.
+    pub fn compute_note_hash(self) -> Field {
+        dep::std::hash::pedersen_hash(serialize(self))
+    }
+    
+    /// @notice Computes the nullifier of the note with context.
+    /// @dev Placeholder for computing the nullifier with context, currently returns the result of `compute_nullifier_without_context`
+    /// @return The computed nullifier of the note.
+    fn compute_nullifier(self, context: &mut PrivateContext) -> Field {
+        let note_hash_for_nullify = compute_note_hash_for_read_or_nullify(PendingCallbackNoteMethods, self);
```

Comment:
We can change all the logic here with this to avoid repetition:

`self.compute_nullifier_without_context()`